README.md

```diff
@@ -42,23 +30,279 @@ Express has extensive support for template rendering and can load and leverage H
+  Notice that in order to serve individual fruit pages we have many static `.html` pages.
 
   <img src="./images-readme/folder-structure-fruits-html.png" width="200px" height="auto"/>
 
-  Notice also that we have two `.html` files to handle errors.
+  Notice also that we have two html files to handle errors.
+
+  Handlebars will help us to avoid duplication by creating template views. In fact Handlebars will allow us to delete all those html files.
+
+### Install handlebars
+
+ 1. In the tasks directory run the follwing code:
+
+    ```bash
+    npm i express-handlebars
+    ```
+
+### Add handlebars to your project
+
+1. Require express-handlebars in app.js:
+
+    ```js
+    const exphbs = require('express-handlebars')
+    ```
+
+2. We now need to tell express the location of the ```views``` folder and the views engine that we want to use:
+    ```js
+    app.set('views', path.join(__dirname, 'views'));
+    app.set('view engine', 'hbs');
+    ```
+
+3. We use ```app.engine``` to tell express to use the view engine (express-handlebars) to render hbs files. We are also configuring express-handlebars by giving the various directories that the engine will search for, and the default layout.
+    ```js
+    app.engine(
+      'hbs',
+      exphbs({
+        extname: 'hbs',
+        layoutsDir: path.join(__dirname, 'views', 'layouts'),
+        partialsDir: path.join(__dirname, 'views', 'partials'),
+        defaultLayout: 'main',
+      })
+    )
+    ```
+  4. We now need to make the relevant folders. Create a ```views``` directory within src. This is what you set in the app.js file as the directory to hold the hbs files.
+  5. In src/views create a ```layouts``` directory.
+  6. In src/views create a ```partials``` directory.
+
+
+### Serving the home page
+
+1. Create a ```home.hbs``` in the views directory
+1. Navigate to public/index.html. Copy all the contents to ```home.hbs```.
+1. Delete public/index.html.
+1. In src/views/layouts create a main.hbs file and insert the following code:
+    ```hbs
+    {{{ body }}} 
+    ```
+    Body will be the hbs file we are rendering.
+
+6. To start the server in development mode use the npm script from the ```package.json``` by typing this in your terminal:
+    ```bash
+    npm run start:watch
+    ```
+
+    Load up the server and go to localhost:3000 in your browser; expect to see a 404 error. We are now going to serve up the ```home.hbs``` file.
+
+7. Create a ```home.js``` in src/controllers
+
+8. In ```home.js``` add the following code:
+    ```js
+    exports.get = (req, res) => {
+      res.render('home');
+    };
+    ```
+    Here ```res.render``` renders a view and sends a rendered HTML string to the client.
+
+9. In src/controllers/index.js require in ```home.js```.
+
+10. Add the following line after files are required in:
+    ``` js
+    router.get('/', home.get);
+    ```
+
+### Split up the contents into partials
```

Comment:
Can we move the addition of partials towards the end (just before the stretch goals)?

Reply:
Not sure I get the reasoning for this suggestion, could you explain?

I think that the way it currently is makes a lot of sense. First the students will put the entire contents of the home page into home.hbs, and will then render this via the main template using {{{body}}}. It seems to me that the natural next step is to create partials, as this is in fact quite similar to factoring out into home.hbs and the main template. Mor importantly, without creating partials at this stage, it's *not clear what the point of the main.hbs template was at all*, or how what we just did achieved anything.

This also sets up the step after that, which has students completing these steps themselves without instructions.

Only after these things do they get on to helpers which I feel are trickier to get to grips with than partials, and should come later.

If the concern is that this step relies too much on the students figuring things out for themselves and that this is too difficult, we could revert to an earlier version of this PR which has step by step instructions. I changed the partials section of the instructions in response to @des-des comment that it was becoming too much like a long list of instructions, and this seemed to me to be the easiest place to get students figuring things out for themselves without getting them completely lost.

Reply:
@BartBucknill ah, I get what you mean. In these instructions it seems to combine the idea of a layout and partials. IMO I would introduce the two functionalities separately as they serve different purposes. For overall flow of the workshop I would:
- setup files
- create layout (with html head and empty body tags)
- create the body (and show it feeds into the layout)
- introduce helpers (as this is a key concept)
- refactor duplicated html into partials

I would leave partials to the end as I see it as syntactic sugar, which isn't obligatory for understanding or setting up handlebars. 

Currently the PR is a huge improvement from what was delivered last time. I'm happy to move this suggestion to an issue to be reviewed after the next cohort.